ms2/src/org/labkey/ms2/metadata/MassSpecFractionsDomainKind.java

```diff
@@ -122,8 +125,19 @@ public void appendNavTrail(NavTree root, Container c, User user)
     }
 
     @Override
-    public Domain createDomain(GWTDomain domain, JSONObject arguments, Container container, User user, @Nullable TemplateInfo templateInfo)
+    public Domain createDomain(GWTDomain domain, SampleTypeDomainKindProperties arguments, Container container, User user, @Nullable TemplateInfo templateInfo)
     {
-        return _assayDelegate.createDomain(domain, arguments, container, user, templateInfo);
+        //TODO this is probably broken.
+        JSONObject args = arguments != null ? arguments.toJSONObject() : null;
+        return _assayDelegate.createDomain(domain, args, container, user, templateInfo);
+    }
+
+    @Override
+    @NotNull
+    public ValidationException updateDomain(GWTDomain<? extends GWTPropertyDescriptor> original, GWTDomain<? extends GWTPropertyDescriptor> update, @Nullable SampleTypeDomainKindProperties options, Container container, User user, boolean includeWarnings)
+    {
+        //TODO this is probably broken too. Maybe we should throw a not supported exception here?
```

Comment:
same comment as above related to this TODO

Reply:
This method isn't called as far as I can tell. This overrides the underlying SampleSetDomainKind, but the implemented Assay gets updated through the AssayService. The better option here I think is throwing a NotImplementedException or similar.